src/main/java/org/diego/tutorial/car/timers/CarsTimer.java

```diff
@@ -0,0 +1,40 @@
+package org.diego.tutorial.car.timers;
+
+import java.util.List;
+
+import javax.ejb.EJB;
+import javax.ejb.Schedule;
+import javax.ejb.Singleton;
+
+import org.apache.log4j.Logger;
+import org.diego.tutorial.car.model.Car;
+import org.diego.tutorial.car.model.service.CarService;
+
+/**
+ * Class that implements a timer for the operations related to the cars
+ * */
+@Singleton
+public class CarsTimer {
+	@EJB
+	private CarService carService;
+	
+	private final static Logger LOGGER = Logger.getLogger(CarsTimer.class);
+	
+	/**
+	 * Method executed every minute, that gets a list of all the 
+	 * non-checked cars, and them checks them.
+	 * */
+	@Schedule(hour = "*", minute = "*/1", persistent = false)
+	public void checkedField() {
+		LOGGER.info("Checking the cars");
+		List<Car> nonCheckedCars = carService.getAllNonCheckedCars();
+		
+		for (Car car : nonCheckedCars) {
+			LOGGER.info("Checking the car: " + car);
+			car.setChecked(true);
+			carService.updateCar(car);
```

Comment:
once setChecked is set to true, is it not automatically persisted to DB by JPA at the end of the transaction without using the service?

Reply:
true, fixed for next commit

Reply:
Depends on autocommit setting. I think it will be set to true by default, but due to performance issues can be found disabled in many production projects

Reply:
In the system it worked without explicitly call the update method again, so I removed it.